x/profiles/types/codec.go

```diff
@@ -83,18 +88,30 @@ func RegisterInterfaces(registry types.InterfaceRegistry) {
 	msgservice.RegisterMsgServiceDesc(registry, &_Msg_serviceDesc)
 }
 
+// ModuleCdc references the global x/profiles protobuf codec. Note, the codec should
+// ONLY be used in certain instances of tests and for JSON encoding.
+//
+// The actual codec used for serialization should be provided to x/profiles and
+// defined at the application level.
+func ModuleCdc() *codec.ProtoCodec {
```

Comment:
Why was this changed from a variable to a method? If we do this, each time we wall `ModuleCdc` we are de-facto re-registering all the interfaces and returning a new coded. Which might impact performance issues. Why wasn't this kept as a variable and an `init` method used instead if needed?

Reply:
The `init` of [models.pb.go ](https://github.com/desmos-labs/desmos/blob/master/x/profiles/types/models_chain_links.pb.go#L420) runs after the codec.go `init`. This causes 
error that the interfaces registration of `x/profiles` is before proto message name registration so the typeURL of `x/profiles` types are always `/`:
```
panic: concrete type *types.Base58Address has already been registered under typeURL /, cannot register *types.HexAddress under same typeURL. This usually means that there are conflicting modules registering different concrete types for a same interface implementation
```

Reply:
I realize that the current app codec is already proto codec which registered all the required modules, so we can use it directly without any problems instead of maintaining another one for `x/profiles`